### ISSUE TITLE
About the pkl file

### ISSUE THREAD

**Issue by jiayunshu** [Chuhanxx/FontAdaptor#10]
I want to use my own data for training, can you tell me how to get the pkl file？

**Comment by Chuhanxx** [comment]
Hi, 

The `train_seg_map.pkl` is for labelling the characters segmentation in training text-line images. It is used for our FontSynth dataset as our text-line images are synthetically generated offline before training. 

However, when using customized data, to minimize the cost of generating and lablelling text-line images, the text-line images are generated by concatenating character images in sequence. In this case, only a set of character images are needed for training. The segmentation ground-truth can be online computed as we know the width of each character image, therefore no explicit pkl file is required. 
There is the function for it: https://github.com/Chuhanxx/FontAdaptor/blob/master/dataloader.py#L246

**Comment by jiayunshu** [comment]
Thank you very much for replying to me, but I find that seg_labels are used when calculating the loss. It can be seen in: https://github.com/Chuhanxx/FontAdaptor/blob/master/train.py#L156
But the seg_label is from the pkl file. It can be seen in: https://github.com/Chuhanxx/FontAdaptor/blob/master/dataloader.py#L83
https://github.com/Chuhanxx/FontAdaptor/blob/master/dataloader.py#L95
How can I get the seg_label?

**Comment by Chuhanxx** [comment]
Hi, 

`train_seg_map.pkl` is basically a pickle file where the ground-truth segmentation of each training text-line is saved. Each text-line has an array to label the boundaries between characters, written as [b0, b1, b2, b3......bn], where the b0 is the x coordinate of the starting point of the 1st character, and b1 is the x coordinate of the boundary between the 1st and 2nd characters, etc.
These arrays are stored as a dictionary in the pickle file which looks like:

> {font_name1:
>      ..{image_name1: array} ,
>      ..{image_name2 :array},
>      ...
> font_name2:
>      ..{image_name1: array},
>      ..{image_name2 :array},
>.     ...
> }


When you are training with your own data, you can either: 1) generate and store a pickle file in this format, and read it directly from your disk in training using our code, 2) use https://github.com/Chuhanxx/FontAdaptor/blob/master/dataloader.py#L246 to generate it online as I mentioned above, although in our source code this function is used to generate `char_seg_labels`, it can also be used for `char_labels` since they follow the same rule.

**Comment by jiayunshu** [comment]
Hello, I'm sorry to bother you again, but I want to know if there is no way to get the seg_label when I don't know the width of each character image in the text-line images.
